lib/ansible/modules/cloud/amazon/ec2_asg.py

```diff
@@ -441,173 +510,272 @@ def create_autoscaling_group(connection, module):
+        region, ec2_url, aws_connect_params = get_aws_connection_info(module, boto3=True)
         try:
-            ec2_connection = connect_to_aws(boto.ec2, region, **aws_connect_params)
-        except (boto.exception.NoAuthHandlerFound, AnsibleAWSError) as e:
+            ec2_connection = boto3_conn(module,
+                                        conn_type='client',
+                                        resource='ec2',
+                                        region=region,
+                                        endpoint=ec2_url,
+                                        **aws_connect_params)
+        except (botocore.exceptions.NoCredentialsError, AnsibleAWSError) as e:
             module.fail_json(msg=str(e))
     elif vpc_zone_identifier:
         vpc_zone_identifier = ','.join(vpc_zone_identifier)
 
     asg_tags = []
     for tag in set_tags:
-        for k,v in tag.items():
+        for k,v in iteritems(tag):
             if k !='propagate_at_launch':
-                asg_tags.append(Tag(key=k,
-                     value=v,
-                     propagate_at_launch=bool(tag.get('propagate_at_launch', True)),
-                     resource_id=group_name))
-
-    if not as_groups:
+                asg_tags.append(dict(Key=k,
+                                     Value=v,
+                                     PropagateAtLaunch=bool(tag.get('propagate_at_launch', True)),
+                                     ResourceType='auto-scaling-group',
+                                     ResourceId=group_name))
+    if not as_groups.get('AutoScalingGroups'):
         if not vpc_zone_identifier and not availability_zones:
             availability_zones = module.params['availability_zones'] = [zone.name for zone in ec2_connection.get_all_zones()]
         enforce_required_arguments(module)
-        launch_configs = connection.get_all_launch_configurations(names=[launch_config_name])
-        if len(launch_configs) == 0:
-            module.fail_json(msg="No launch config found with name %s" % launch_config_name)
-        ag = AutoScalingGroup(
-                 group_name=group_name,
-                 load_balancers=load_balancers,
-                 availability_zones=availability_zones,
-                 launch_config=launch_configs[0],
-                 min_size=min_size,
-                 max_size=max_size,
-                 placement_group=placement_group,
-                 desired_capacity=desired_capacity,
-                 vpc_zone_identifier=vpc_zone_identifier,
-                 connection=connection,
-                 tags=asg_tags,
-                 health_check_period=health_check_period,
-                 health_check_type=health_check_type,
-                 default_cooldown=default_cooldown,
-                 termination_policies=termination_policies)
+        launch_configs = connection.describe_launch_configurations(LaunchConfigurationNames=[launch_config_name])
```

Comment:
Fixed this on the line below.
describe_launch_configurations() is returning something like 
```
{'LaunchConfigurations': [], 'ResponseMetadata': {'RetryAttempts': 0, 'HTTPStatusCode': 200, 'RequestId': '57dccc7c-2080-11e7-8fbd-adf472434545', 'HTTPHeaders': {'x-amzn-requestid': '57dccc7c-2080-11e7-8fbd-adf472434545', 'date': 'Thu, 13 Apr 2017 19:35:25 GMT', 'content-length': '350', 'content-type': 'text/xml'}}}"
```
So it always had longer than a length of 0 and the below check had no effect. Added a check for the same thing on lines 719-720 too.